content/doc/book/pipeline/syntax.adoc

```diff
@@ -0,0 +1,851 @@
+each `stage` directive.
+
+any:: Execute the Pipeline, or stage, on any available agent. For example: `agent any`
+none:: When applied at the top-level of the `pipeline` block, each `stage`
+directive will need to contain its own `agent` directive. For example: `agent none`
+label:: Execute the Pipeline, or stage, on the agent available in the Jenkins
+environment with the provided label. For example: `agent { label 'my-defined-label' }`
+docker:: Execute the Pipeline, or stage, with the given container which will be
+dynamically provisioned on a <<../glossary#node, node>> pre-configured to
+accept Docker-based Pipelines, or a node matching the optionally defined
+`label` parameter.  The `docker` also optionally accepts an `args` parameter
+which may contain parameters to pass directly to a `docker run` invocation.
+ For example: `agent { docker: 'maven:3-alpine' }` or
+[source,groovy]
+----
+agent {
+    docker {
+        image: 'maven:3-alpine'
+        label: 'my-defined-label'
+        args:  '-v /tmp:/tmp'
+    }
+}
+----
+
+////
+XXX: https://issues.jenkins-ci.org/browse/JENKINS-41607
+
+dockerfile:: Execute the Pipeline, or stage, with a container built from a
+`Dockerfile` contained in the source repository. Conventionally this can be the
+`Dockerfile` in the root of the source repository: `agent { dockerfile true }`
+
+////
+
+
+
+[[agent-example]]
+===== Example
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent { docker 'maven:3-alpine' } /* <1> */
+    stages {
+        stage('Example Build') {
+            steps {
+                sh 'mvn -B clean verify'
+            }
+        }
+    }
+}
+----
+<1> Execute all the steps defined in this Pipeline within a running container
+of the given name and tag (`maven:3-alpine`).
+
+====== Stage-level `agent` directive
+
+[pipeline]
+----
+// Declarative //
+pipeline {
+    agent none /* <1> */
+    stages {
+        stage('Example Build') {
+            agent { docker: 'maven:3-alpine' } /* <2> */
+            steps {
+                sh 'mvn -B clean verify'
+            }
+        }
+        stage('Example Test') {
+            agent { docker: 'openjdk:8-jre' } /* <3> */
+            steps {
+                echo 'Hello JRE'
+            }
+        }
+    }
+}
+----
+<1> Defining `agent none` at the top-level of the Pipeline ensures that
+<<../glossary#executor, an Executor>> will not be unnecessarily. Using `agent
+none` requires that each `stage` directive contain an `agent` directive.
+<2> Execute the steps contained within this stage using the given container.
+<3> Execute the steps contained within this steps using a different container
+from the previous stage.
+
+
+==== environment
+
+The `environment` directive specifies a sequence of key-value pairs which will
+be defined as environment variables for the all steps, or stage-specific steps,
+depending on where the `environment` directive is located within the Pipeline.
```

Comment:
Once somebody writes down what the allowed names are for me, I'll gladly include them :smile: